waiter/test/waiter/process_request_test.clj

```diff
@@ -307,47 +278,47 @@
                {:authorization/user "test-user"}
-               {:status 202, :headers {"location" "../result/location"}}))))
+               {:status 202 :headers {"location" "../result/location"}}))))
     (testing "202-relative-location-2"
-      (is (= {:auth-user "test-user", :location "/query/for/result/location", :result :success-async}
+      (is (= {:auth-user "test-user" :location "/query/for/result/location" :query-string nil :result :success-async}
              (execute-inspect-for-202-async-request-response
                "http://www.example.com:1234/query/for/status"
                {:authorization/user "test-user"}
-               {:status 202, :headers {"location" "result/location"}}))))
+               {:status 202 :headers {"location" "result/location"}}))))
     (testing "202-relative-location-two-levels"
-      (is (= {:auth-user "test-user", :location "/result/location", :result :success-async}
+      (is (= {:auth-user "test-user" :location "/result/location" :query-string "a=b&c=d|e" :result :success-async}
              (execute-inspect-for-202-async-request-response
                "http://www.example.com:1234/query/for/status"
-               {:authorization/user "test-user"}
-               {:status 202, :headers {"location" "../../result/location"}}))))
+               {:authorization/user "test-user" :query-string "a=b&c=d|e"}
+               {:status 202 :headers {"location" "../../result/location"}}))))
     (testing "202-absolute-url-same-host-port"
-      (is (= {:auth-user "test-user", :location "/retrieve/result/location", :result :success-async}
+      (is (= {:auth-user "test-user" :location "/retrieve/result/location" :query-string nil :result :success-async}
              (execute-inspect-for-202-async-request-response
                "http://www.example.com:1234/query/for/status"
                {:authorization/user "test-user"}
-               {:status 202, :headers {"location" "http://www.example.com:1234/retrieve/result/location"}}))))
+               {:status 202 :headers {"location" "http://www.example.com:1234/retrieve/result/location"}}))))
     (testing "202-absolute-url-different-host"
       (is (= {:result :not-async}
              (execute-inspect-for-202-async-request-response
                "http://www.example.com:1234/query/for/status"
                {:authorization/user "test-user"}
-               {:status 202, :headers {"location" "http://www.example2.com:1234/retrieve/result/location"}}))))
+               {:status 202 :headers {"location" "http://www.example2.com:1234/retrieve/result/location"}}))))
     (testing "202-absolute-url-different-port"
       (is (= {:result :not-async}
              (execute-inspect-for-202-async-request-response
                "http://www.example.com:1234/query/for/status"
                {:authorization/user "test-user"}
-               {:status 202, :headers {"location" "http://www.example.com:5678/retrieve/result/location"}}))))))
```

Comment:
It would be easier to review this PR (handling the query string) if the unrelated refactoring changes (e.g., removing the commas here) were in a separate PR.

Reply:
Sorry, will try and avoid next time.